drivers/hmis/app/models/hmis/hoh_change_handler.rb

```diff
@@ -0,0 +1,114 @@
+module Hmis
+  #
+  # The HohChangeHandler class is responsible for handling changes to the Head of Household (HoH) in a household enrollment.
+  # It provides methods to perform the change, apply the changes to the database, and validate the changes.
+  #
+  # This logic is split out from the UpdateRelationshipToHoH mutation in case we need to use it in cleanup routines,
+  # for example bulk assigning HoH to the oldest adult in a household.
+  #
+  # Parameters:
+  # - new_hoh_enrollment: The Enrollment that is becoming the Head of Household
+  # - hud_user_id: The "UserID" of the HUD user making the change.
+  class HohChangeHandler
+    attr_accessor :new_hoh_enrollment, :household_enrollments, :old_hoh_enrollments, :new_hoh_move_in_date, :old_hoh_move_in_date, :validation_errors, :hud_user_id
+
+    def initialize(new_hoh_enrollment:, hud_user_id:)
+      self.new_hoh_enrollment = new_hoh_enrollment
+      self.hud_user_id = hud_user_id
+
+      # All Enrollments in the household (including new and old HoHs)
+      self.household_enrollments = new_hoh_enrollment.household_members.preload(:client)
+      # Previous HoHs (should only be one, but we handle multiple for data cleanup)
+      self.old_hoh_enrollments = household_enrollments.
+        filter(&:head_of_household?).
+        sort_by { |en| [en.DateCreated, en.id] }
+
+      self.old_hoh_move_in_date = old_hoh_enrollments.map(&:move_in_date).compact.first
+      self.new_hoh_move_in_date = [old_hoh_move_in_date, new_hoh_enrollment.entry_date].max if old_hoh_move_in_date
```

Comment:
Since this is the move-in date for the household, I think you want the min, and might possibly have to move the new HoH entry date back so that they are present for the entire household enrollment.  That's what we're supposed to do for reporting, but there has been acknowledgment that it doesn't reflect reality.  Relatedly, annual assessments are on the anniversary of the HoH entry date, so if you change that, previous annual assessments will have been corrected outside of the window and future assessments will be on a different cadence.

Reply:
responded on issue https://github.com/open-path/Green-River/issues/6857#issuecomment-2437677775

Reply:
OK, https://github.com/greenriver/hmis-warehouse/pull/4852/commits/e38d819dac5dbca42638a7d5ba0b147bb0b9d54a should reflect the change discussed at standup. We do NOT transfer Move-in Date in the scenario where the new HoH entered after move-in. This ends up in a "bad data" scenario, because the non-HOh still has a move-in date. Plan to adjust pending HUD guidance, so left a bunch of TODOs.